app/views.py

```diff
@@ -102,12 +103,19 @@ def choose_transcription(request: HttpRequest) -> HttpResponse:
     if options.count() > 3:
         # we have more to choose from, so let's grab 3
         temp = []
-        while len(temp) < 3:
-            submission = random.choice(options)
+        options_list = list(options)
+        for _ in range(options.count()):
+            if len(temp) >= 3:
+                break
+            # What if the only options we have here are video options? This will
+            # terminate when it runs out of options to try but we at least get
+            # to inject a little randomness into it.
+            submission = random.choice(options_list)
             # todo: we want to eventually handle other types of content here, but
             #  we simply aren't ready for it yet.
-            if submission not in temp and submission.is_image:
-                temp.append(submission)
+            if submission.is_image:
+                temp.append(options_list.pop(options_list.index(submission)))
+
```

Comment:
Here I would filter the list by `submission.is_image` and then only run the loop over `range(0, max(3, options.count())`. That way we can guarantee termination (and we currently fetch all options anyway, so it's probably similar for efficiency).

Reply:
Yep yep that makes total sense to me. I'll fix it up.